commands/audit/sca/pnpm/pnpm.go

```diff
@@ -46,10 +48,21 @@ func BuildDependencyTree(params utils.AuditParams) (dependencyTrees []*xrayUtils
 		return
 	}
 	// Build
-	if err = installProjectIfNeeded(pnpmExecPath, currentDir); errorutils.CheckError(err) != nil {
+	var dirForDependenciesCalculation string
+	if dirForDependenciesCalculation, err = installProjectIfNeeded(pnpmExecPath, currentDir); errorutils.CheckError(err) != nil {
 		return
 	}
-	return calculateDependencies(pnpmExecPath, currentDir, params)
+
+	if dirForDependenciesCalculation == "" {
```

Comment:
Is this case still relevant?

Reply:
yes because if we do not need to install we will get an empty value and the directory on which we need to calculate the dependencies is the original directory.
In this case the original dir will not be effected so we are all good